contracts/toCKB-typescript/src/switch/mod.rs

```diff
@@ -132,6 +136,34 @@ fn get_deletion_tx_type(data: &ToCKBCellDataView) -> Result<TxType, Error> {
     }
 }
 
+fn verify_xt(tx_type: &TxType) -> Result<(), Error> {
+    use TxType::*;
+    match tx_type {
+        DepositRequest
+        | Bonding
+        | WithdrawPledge
+        | WithdrawCollateral
+        | WithdrawPledgeAndCollateral
+        | LiquidationSignerTimeout
+        | LiquidationUndercollateral
+        | LiquidationFaultyWhenWarranty
+        | LiquidationFaultyWhenRedeeming => forbid_mint_xt(),
+        _ => Ok(()),
+    }
+}
+
+fn forbid_mint_xt() -> Result<(), Error> {
+    let sudt_cell_count = QueryIter::new(load_cell_type, Source::Output)
+        .filter(|type_opt| type_opt.is_some())
+        .map(|type_opt| type_opt.unwrap())
+        .filter(|script| script.code_hash().raw_data().as_ref() == SUDT_CODE_HASH.as_ref())
+        .count();
+    if 0 != sudt_cell_count {
```

Comment:
Should we also add input check? The code now allows burn xt, which may not be vulnerable, but not expected.

Reply:
no need, we can't prevent user from burning tokens anyhow